packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -190,11 +201,216 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const [isMineOnly, setIsMineOnly] = useState(false);
+    const [searchText, setSearchText] = useState('');
+    const closeQueryImporter = (): void => {
+      editorStore.setIsQueryLoaderOpen(false);
+    };
+    const handleEnterQueryImporter = (): void =>
+      queryFinderRef.current?.focus();
+    const loadSelectedQuery = (): void => {
+      if (selectedQueryID) {
+        editorStore.setIsQueryLoaderOpen(false);
+        applicationStore.navigator.goTo(
+          generateExistingQueryEditorRoute(selectedQueryID),
+        );
+        window.location.reload();
+        applicationStore.setBlockingAlert({
+          message: 'Loading query...',
+          showLoading: true,
+        });
+      } else {
+        applicationStore.notifyWarning(
+          'No query is selected. Please select a query to load or close.',
+        );
+      }
+    };
+    const searchInputRef = useRef<HTMLInputElement>(null);
+    const loadQueries = (text: string): void => {
+      flowResult(editorStore.loadQueries(text))
+        .then(() => editorStore.loadQueriesState.pass())
+        .catch(applicationStore.alertUnhandledError);
+    };
+    const searchQuery: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+      setSearchText(event.target.value);
+      loadQueries(event.target.value);
+    };
```

Comment:
Isn't this expensive? You can just set `queries = []`, this will make another network call!

Reply:
By setting queries = [], it will make table empty, which is not consistent with what we will show when the query loader is open.
I could create a new vairable to store the first 10 (QUERY_BUILDER_QUERY_LOAD_MAX_AMOUNT) queries to avoid another network call.

Reply:
I see your point, so let's query for `''` then, don't store the first 10 queries

Reply:
gotcha. I changed it back. thanks!